concordium-consensus/src/Concordium/KonsensusV1/Scheduler.hs

```diff
@@ -373,13 +378,23 @@ processBlockRewards ::
     ParticipatingBakers ->
     -- | Transaction fees and number of "free" transactions.
     TransactionRewardParameters ->
+    [(BakerId, Word64)] ->
     -- | Block state.
     UpdatableBlockState m ->
     m (UpdatableBlockState m)
-processBlockRewards ParticipatingBakers{..} TransactionRewardParameters{..} theState0 = do
-    theState1 <- bsoNotifyBlockBaked theState0 pbBlockBaker
-    theState2 <- bsoMarkFinalizationAwakeBakers theState1 pbQCSignatories
-    doBlockRewardP4 trpTransactionFees trpFreeTransactionCounts pbBlockBaker theState2
+processBlockRewards ParticipatingBakers{..} TransactionRewardParameters{..} missedRounds theState0 = do
+    -- Note: The order of the following operations is important.
+    --
+    -- The finalization awake bakers are the ones that signed the QC (possibly
+    -- a timeout) for the parent block. If it was a timeout, missed rounds are counted
+    -- between this last timeout round and the round of the block that is baked next. So if
+    -- a validator signed the QC for the parent block, but missed a round in the interim,
+    -- then its missed round counter would be 1. Though if it also baked the new block, it
+    -- would be reset to 0.
+    theState1 <- bsoMarkFinalizationAwakeBakers theState0 pbQCSignatories
+    theState2 <- bsoUpdateMissedRounds theState1 missedRounds
```

Comment:
This should probably only be done if the protocol version supports suspending inactive validators.

Reply:
As above, this is already a no-op if validator suspension is not supported by the protocol version. I added however an explicit check here now.